### ISSUE TITLE
[bug] `prefer-destructuring-assignment` detects unrelated code as components

### ISSUE THREAD

**Issue by ulrichstark** [Rel1cx/eslint-react#1488]
The lint rule `prefer-destructuring-assignment` detects unrelated code as components. I only had the first (`flatMap`) case in our project, but noticed there's at least the same issue with `filter` and `find`, but not with `map`.

```tsx
const items = [{ property: "value" }];

items.flatMap((item) => {
  console.log(item.property); // Reports "Use destructuring assignment for component props"
  return null;
});

items.filter((item) => {
  console.log(item.property); // Reports "Use destructuring assignment for component props"
  return null;
});

items.find((item) => {
  console.log(item.property); // Reports "Use destructuring assignment for component props"
  return null;
});

items.map((item) => {
  console.log(item.property); // Doesn't report
  return null;
});
```

**Comment by ulrichstark** [comment]
Thank you very much! After #1472, #1476 and now this issue closed, we are now finally able to upgrade to the most recent version without any false positives. Now there's only #1426 left to be fixed so I can give you something back :)
